src/bin/ch-remote.rs

```diff
@@ -652,334 +895,238 @@ fn create_data(path: &str) -> Result<String, Error> {
-/// Resize a memory zone
-struct ResizeZoneSubcommand {
-    #[argh(option, long = "id")]
-    /// memory zone identifier
-    id: String,
-
-    #[argh(option, long = "size")]
-    /// new memory size in bytes (supports K/M/G suffix)"
-    size: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "snapshot")]
-/// Create a snapshot from VM
-struct SnapshotSubcommand {
-    #[argh(positional)]
-    /// destination_url
-    snapshot_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "restore")]
-/// Restore VM from a snapshot
-struct RestoreSubcommand {
-    #[argh(positional)]
-    /// restore config
-    restore_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "coredump")]
-/// Create a coredump from VM
-struct CoredumpSubcommand {
-    #[argh(positional)]
-    /// coredump config
-    coredump_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "send-migration")]
-/// Initiate a VM migration
-struct SendMigrationSubcommand {
-    #[argh(switch, long = "local")]
-    /// local migration
-    send_migration_local: bool,
-
-    #[argh(positional)]
-    /// destination_url
-    send_migration_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "receive-migration")]
-/// Receive a VM migration
-struct ReceiveMigrationSubcommand {
-    #[argh(positional)]
-    /// receiver url
-    receive_migration_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "create")]
-/// Create a VM from a JSON configuration
-struct CreateSubcommand {
-    #[argh(positional, default = "String::from(\"-\")")]
-    /// vm config
-    vm_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "version")]
-/// Print version information
-struct VersionSubcommand {}
-
 fn main() {
-    let toplevel: TopLevel = argh::from_env();
+    let app = Command::new("ch-remote")
+        .author(env!("CARGO_PKG_AUTHORS"))
+        .subcommand_required(true)
+        .about("Remotely control a cloud-hypervisor VMM.")
+        .args([
+            Arg::new("api-socket")
+                .long("api-socket")
+                .help("HTTP API socket path (UNIX domain socket).")
+                .num_args(1),
+            #[cfg(feature = "dbus_api")]
+            Arg::new("dbus-service-name")
+                .long("dbus-service-name")
+                .help("Well known name of the dbus service")
+                .num_args(1),
+            #[cfg(feature = "dbus_api")]
+            Arg::new("dbus-object-path")
+                .long("dbus-object-path")
+                .help("Object path which the interface is being served at")
+                .num_args(1),
+            #[cfg(feature = "dbus_api")]
+            Arg::new("dbus-system-bus")
+                .long("dbus-system-bus")
+                .action(clap::ArgAction::SetTrue)
```

Comment:
Need `.num_args(0)` to enforce no argument for this comamnd.

Also remove `clap::` prefix.